File: server/GameLibraryAPI/GameLibraryAPI/Dto/GameDto.cs
using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
using Microsoft.EntityFrameworkCore.Storage.ValueConversion.Internal;
using System.Text.Json.Serialization;

namespace GameLibraryAPI.Dto
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum PlatformType
    {
        PC,
        PS,
        XBOX
    }
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum FormatType
    {
        DIGITAL,
        DISK
    }
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum CurrencyType
    {
        EUR,
        DOL,
        RON
    }
    public class GameDto
    {
        public int? Id { get; set; }
        public string? Name { get; set; }
        public string? Producer { get; set; }
        public string? PurchaseDate { get; set; }
        public string? LastTimePlayed { get; set; }
        public double? Price { get; set; }
        public CurrencyType? Currency { get; set; }
        public PlatformType? Platform { get; set; }
        public FormatType? FormatOfGame { get; set; }
        public string? PicturePath { get; set; }
    }
}

File: android/app/src/main/java/com/example/gamecollection/data/remote/dto/GameDto.kt
package com.example.gamecollection.data.remote.dto

import com.example.gamecollection.data.local.Currency
import com.example.gamecollection.data.local.Format
import com.example.gamecollection.data.local.Platform
import java.time.LocalDate
import java.time.LocalDateTime

data class GameDto (
    val id: Int?,
    val name: String,
    val producer: String,
    val purchaseDate: String,
    val lastTimePlayed: String,
    val price: Double,
    val currency: Currency,
    val platform: Platform,
    val formatOfGame: Format,
    val picturePath: String,
)